### PR TITLE
[FEATURE] added getMZ as convinience method to AASequence

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Add convenience function for mz calculation to AASequence.

# Checklist:
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [X] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

**Comment by timosachsenberg** [comment]
addresses https://github.com/OpenMS/OpenMS/issues/5016

**Comment by timosachsenberg** [comment]
ready for review. please squash merge if done

**Review by jpfeuffer — APPROVED** [review]

File: src/openms/source/CHEMISTRY/AASequence.cpp [hunk 1]
```diff
@@ -484,6 +484,12 @@ namespace OpenMS
     return tag_offset + getFormula(type, charge).getAverageWeight();
   }
 
+  double AASequence::getMZ(Int charge, Residue::ResidueType type) const
+  {
+    if (charge == 0) throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Can't calculate mass-to-charge ratio for charge=0.", toString());
```
Comment (cbielow) [inline src/openms/source/CHEMISTRY/AASequence.cpp:489]: add `@throws Exception::InvalidValue if charge==0` to docs

File: src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp [hunk 2]
```diff
@@ -1986,7 +1986,7 @@ namespace OpenMS
       // set precursor information
       vector<Precursor> precursors;
       Precursor precursor;
-      precursor.setMZ(aa_sequence.getMonoWeight());
+      precursor.setMZ(aa_sequence.getMZ(charge));
```
Comment (cbielow) [inline src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp:1989]: no the same, is it?
Was it a bug before?

Comment (timosachsenberg) [reply src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp:1989]: it was not used but getMZ(charge) is the correct precursor m/z so it was kind of a bug

File: src/tests/class_tests/openms/source/AASequence_test.cpp [hunk 3]
```diff
@@ -406,6 +406,17 @@ START_SECTION((double getMonoWeight(Residue::ResidueType type = Residue::Full, I
...
 END_SECTION
 
+START_SECTION((double getMZ(Int charge, Residue::ResidueType type = Residue::Full) const))
+{
+  TOLERANCE_ABSOLUTE(1e-6)
+  TOLERANCE_RELATIVE(1.0 + 1e-6)
+  
+  // uses getMonoWeight and is thus thoroughly tested
+  TEST_REAL_SIMILAR(AASequence::fromString("DFPIANGER").getMZ(1, Residue::YIon), double(1018.4952))
+  TEST_REAL_SIMILAR(AASequence::fromString("DFPIANGER").getMZ(2, Residue::YIon), double((1018.4952 + Constants::PROTON_MASS_U) / 2.0))
```
Comment (cbielow) [inline src/tests/class_tests/openms/source/AASequence_test.cpp:416]: test the exception for charge==0?

**Pull request merged**